source/model/Activator.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+
+
+class Activator(ABC):
+
+    def __init__(self, id):
+        self.activatorId = id
+
+    @property
+    @abstractmethod
+    def name(self):
```

Comment:
If you use getters, it is more clear to give them names as get_name. A method should convey the operation it performs.
The same goes for similar cases in this code.